lib/control-connection.js

```diff
@@ -415,6 +415,26 @@ class ControlConnection extends events.EventEmitter {
     }
   }
 
+  async _refreshControlConnection(hostIterator) {
+
+    try {
+
+      this.connection = this._borrowAConnection(hostIterator);
+    }
+    catch(err) {
+
+      /* NODEJS-632: refresh nodes before getting hosts for reconnect since some hostnames may have
+       * shifted during the flight. */
+      this.log("info", "ControlConnection could not reconnect using existing connections.  Refreshing contact points and retrying");
+      this._contactPoints.clear();
+      this._resolvedContactPoints.clear();
+      await Promise.all(this.options.contactPoints.map(name => this._parseContactPoint(name)));
```

Comment:
Does `_parseContactPoint()` stuff the result into `_contactPoints`?

Reply:
Followup question, what happens if that fails? What's the end result in `_contactPoints` that we're going to use in the next line?